src/wallet/direct_send.rs

```diff
@@ -133,12 +142,29 @@ impl Wallet {
             }
         };
 
-        let mut output = Vec::<TxOut>::new();
-
         let txout = {
             let value = match send_amount {
-                SendAmount::Max => (total_input_value - fee).to_sat(),
-                SendAmount::Amount(a) => a.to_sat(),
+                SendAmount::Max => {
+                    // Find dust value for destination scriptpubkey.
+                    let destination_dust = dest_addr.script_pubkey().minimal_non_dust();
+                    if total_input_value > fee + destination_dust {
+                        (total_input_value - fee).to_sat()
+                    } else {
+                        // TODO: Write error statement here.
+                        return Err(WalletError::Protocol("".to_string()));
+                    }
+                }
+                SendAmount::Amount(a) => {
+                    if a + fee <= total_input_value {
+                        a.to_sat()
+                    } else {
+                        return Err(WalletError::Protocol(
+                            "Insufficient funds. The total amount to send plus fees exceeds your available UTXOs. 
+                            Only Swapcoin and SeedCoin UTXOs can be spent."
+                                .to_string(),
```

Comment:
This logic is missplaced. Instead of checking for enough input here we should check after the `total_input_value` step above this section.

At that stage itself the function should fail and return.

A more helpful error message will be putting the `required: amount` and `available: amount` fields in the error message, so user can see the deficit and fund the wallet.

Reply:
The old logic here can stay too.